climada_petals/entity/exposures/osm_dataloader.py

```diff
@@ -0,0 +1,346 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Download openstreetmap data from the Overpass API
+"""
+
+import itertools
+import logging
+import time
+import geopandas as gpd
+import overpy
+import shapely
+
+LOGGER = logging.getLogger(__name__)
+
+class OSMApiQuery:
+    """
+    Queries features directly via the overpass turbo API.
+
+    area: tuple (xmin, ymin, xmax, ymax), list [xmin, ymin, xmax, ymax]
+        or shapely.geometry.Polygon
+    query: str
+        must be of format '["key"]' or '["key"="value"]', etc.
+    """
+
+    def __init__(self, area, condition):
+        self.area = self._area_to_queryformat(area)
+        self.condition = condition
+
+    def _area_to_queryformat(self, area):
+        """
+        reformat lat/lon info as in OSM convention, meaning
+        bbox: (S,W,N,E) instead of (xmin, ymin, xmax, ymax)
+        Points: lat / lon instead of (x,y)
+        """
+        if isinstance(area,(tuple, list)):
+            xmin, ymin, xmax, ymax = area
+            return (ymin, xmin, ymax, xmax)
+
+        if isinstance(area,shapely.geometry.Polygon):
+            lon, lat = area.exterior.coords.xy
+            lat_lon_str = " ".join([str(y)+" "+str(x) for y, x in zip(lat, lon)])
+            return f'(poly:"{lat_lon_str}")'
+    
+        return None
+
+    def _insistent_osm_api_query(self, query_clause, read_chunk_size=100000,
```

Comment:
Jenkins has a point: This method uses nothing in `self`, so it could become a "free" function instead

Reply:
true. since there were many functions organised in 4 classes in this module initially, it made sense to put this in its parent class for organisational reasons. there are still a few more of these. will make sense to restructure / entirely remove the OSMApiQuery class in the future.